src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -155,6 +156,12 @@ static ADUC_Result Script_Handler_DownloadPrimaryScriptFile(ADUC_WorkflowHandle
     try
     {
         result = ExtensionManager::Download(&entity, handle, &Default_ExtensionManager_Download_Options, nullptr);
+        if (IsAducResultCodeSuccess(result.ResultCode))
+        {
+            // Set execute permissions on the downloaded script file
+            std::string scriptPath = std::string(workFolder) + "/" + scriptFileName;
+            chmod(scriptPath.c_str(), S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
```

Comment:
It appears that this code may be redundant, as file permissions are managed here:

https://github.com/Azure/iot-hub-device-update/blob/663b7b700b1c98fce2c4cdf03e3a29f942a86568/src/adu-shell/src/script_tasks.cpp#L47

The original error might have been due to the adu-shell folder not being specified.
Cout your try to test the scenario again without this change again please?